sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample06_TrackWhenAPartitionIsClosed.cs

```diff
@@ -0,0 +1,247 @@
+                                   string blobContainerName)
+        {
+            // When the Event Processor client begins processing, it will take ownership over a set of Event Hub partitions to process.  The ownership that a
+            // processor has of a partition is transient; ownership may be relinquished by the processor in several scenarios:
+            //
+            // - Because processors work collaboratively within the context of a consumer group, they will share responsibility for partitions of the Event Hub,
+            //   with ownership of a partition potentially moving from one processor to another.
+            //
+            // - If a processor encounters an issue and believes that it cannot safely recover processing for a partition that it has ownership of, it will
+            //   relinquish ownership and allow the partition to be claimed by another processor or may reclaim it with fresh state itself.  This scenario will
+            //   also result in errors being sent for processing by the handler.
+            //
+            // - Should there be a request for a processor to stop processing, it will relinquish ownership of its partitions to allow for other processors to
+            //   potentially claim them so that processing continues as event processor instances are scaled for the consumer group.
+            //
+            // In any of these cases, the "PartitionClosing" event will be triggered on the processor.  While there are no actions that you may take to influence
+            // the closing of a partition, this event serves as the logical partner of the "PartitionInitializing" event, allowing you to track the partitions owned
+            // by the processor.
+
+            string consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+            BlobContainerClient storageClient = new BlobContainerClient(blobStorageConnectionString, blobContainerName);
+            EventProcessorClient processor = new EventProcessorClient(storageClient, consumerGroup, eventHubsConnectionString, eventHubName);
+
+            // For this example, we'll keep a list of partitions that are owned by our processor instance and keep it up to
+            // date with the last event processed for that partition.  When our processor closes that partition, we'll ensure that we
+            // have an up-to-date checkpoint based on the last event.
+
+            int eventsProcessed = 0;
+            ConcurrentDictionary<string, ProcessEventArgs> ownedPartitions = new ConcurrentDictionary<string, ProcessEventArgs>();
+
+            // The handler for partition initialization begin tracking the partition.
+
+            Task partitionInitializingHandler(PartitionInitializingEventArgs eventArgs)
+            {
+               try
+               {
+                  ownedPartitions.AddOrUpdate(eventArgs.PartitionId, default(ProcessEventArgs), (_partition, _event) => default(ProcessEventArgs));
```

Comment:
We could just make it `ownedPartitions[eventArgs.PartitionId] = default(ProcessEventArgs)` if we are overwriting it anyway.

Reply:
We could do the same thing in the process event handler.

Reply:
Yeah.... I'm probably trying too hard to make it clear that this is a concurrent dictionary.   Took the suggestion and simplified.